test/LetsTrace.Jaeger.Tests/Serialization/JaegerThriftSerializationTests.cs

```diff
@@ -108,65 +107,61 @@ public void BuildJaegerLog()
             Assert.Equal(1518780809000000, converted.Timestamp);
 
             Assert.Equal(JaegerTagType.DOUBLE, converted.Fields[0].VType);
-            Assert.Equal(doubleField.Key, converted.Fields[0].Key);
-            Assert.Equal(doubleField.Value, converted.Fields[0].VDouble);
+            Assert.Equal(doubleFieldKey, converted.Fields[0].Key);
+            Assert.Equal(fields[doubleFieldKey], converted.Fields[0].VDouble);
 
             Assert.Equal(JaegerTagType.DOUBLE, converted.Fields[1].VType);
-            Assert.Equal(decimalField.Key, converted.Fields[1].Key);
-            Assert.Equal((double)decimalField.Value, converted.Fields[1].VDouble);
+            Assert.Equal(decimalFieldKey, converted.Fields[1].Key);
+            Assert.Equal(Convert.ToDouble(fields[decimalFieldKey]), converted.Fields[1].VDouble);
 
             Assert.Equal(JaegerTagType.BOOL, converted.Fields[2].VType);
-            Assert.Equal(boolField.Key, converted.Fields[2].Key);
-            Assert.Equal(boolField.Value, converted.Fields[2].VBool);
+            Assert.Equal(boolFieldKey, converted.Fields[2].Key);
+            Assert.Equal(fields[boolFieldKey], converted.Fields[2].VBool);
 
             Assert.Equal(JaegerTagType.LONG, converted.Fields[3].VType);
-            Assert.Equal(uint16Field.Key, converted.Fields[3].Key);
-            Assert.Equal(uint16Field.Value, converted.Fields[3].VLong);
+            Assert.Equal(uint16FieldKey, converted.Fields[3].Key);
+            Assert.Equal(Convert.ToInt64(fields[uint16FieldKey]), converted.Fields[3].VLong);
```

Comment:
Do you even have to use `Convert.ToInt64`? Doesn't the `Assert.Equal` statement also work without it?

Reply:
yeah, the `Assert.Equal` throws saying `Expected: 5 (System.UInt16) Actual:   5 (System.Int64)`. Unless you were referring to line 143 where I actually didn't need it.

Reply:
ok, makes sense. thanks for the response!